src/main/java/com/fasterxml/jackson/core/json/ReaderBasedJsonParser.java

```diff
@@ -740,6 +740,10 @@ public final JsonToken nextToken() throws IOException
 
         // Ok: we must have a value... what is it?
 
+        if (i == '+' && isEnabled(JsonReadFeature.ALLOW_LEADING_PLUS_SIGN_FOR_NUMBERS.mappedFeature())) {
+            return nextToken();
```

Comment:
Ok, this is problematic as it allows something like `++++4` I think. Or, potentially, DoS with 10,000 plus signs. :)

But I think it should be possible to handle this as a new entry in `switch` below, maybe adding case right above `0` ... `9`, reading next character, falling through?
Same goes for all parsers.